### PR TITLE
Raise error if torch.compile is called from windows or py 3.11

### PR DESCRIPTION
**Pull request by williamwen42**
For https://github.com/pytorch/pytorch/issues/94914

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Sounds good to me. This will need to be cherry picked (instructions at https://github.com/pytorch/pytorch/issues/94937)

FYI @ezyang in case you weren't pinged here

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -413,13 +413,9 @@ def toy_example(a, b):
     if disable or os.environ.get("TORCHDYNAMO_DISABLE", "") == "1":
         return _NullDecorator()
     if sys.platform == "win32":
-        warnings.warn(
-            "Windows is not currently supported, torch.compile() will do nothing"
-        )
-        return _NullDecorator()
+        raise RuntimeError("Windows not yet supported for torch.compile")
```
Comment (Skylion007) [inline torch/_dynamo/eval_frame.py:416]: Nit: Should be NotImplementedError?

File: torch/_dynamo/eval_frame.py [hunk 2]
```diff
@@ -413,13 +413,9 @@ def toy_example(a, b):
...
     if sys.version_info >= (3, 11):
-        warnings.warn("Python 3.11+ not yet supported, torch.compile() will do nothing")
-        return _NullDecorator()
+        raise RuntimeError("Python 3.11+ not yet supported for torch.compile")
```
Comment (Skylion007) [inline torch/_dynamo/eval_frame.py:418]: Should also be NotImplementedError?

File: torch/_dynamo/eval_frame.py (hunk 1 above)
Comment (williamwen42) [reply torch/_dynamo/eval_frame.py:416]: I would imagine NotImplementedError is typically used for unimplemented methods in abstract base classes, not for missing features.

**Comment by Raghvender1205** [comment]
I have installed the latest version `2.0.0+cu117` and I am getting this error

```
RuntimeError: Windows not yet supported for torch.compile
```

Does pytorch still not support `torch.compile` on windows

**Pull request closed without merging**